src/transport/SecureSessionTable.cpp

```diff
@@ -0,0 +1,257 @@
+    for (auto *session = sortableSessions.Get(); session != (sortableSessions.Get() + numSessions); session++, i++) {
+        ChipLogDetail(SecureChannel, "\t%d: [%p] -- State: '%s', ActivityTime: %lld",
+                     i, session->mSession, session->mSession->GetStateStr(), session->mSession->GetLastActivityTime().count()); 
+    }
+#endif
+
+    for (auto *session = sortableSessions.Get(); session != (sortableSessions.Get() + numSessions); session++) {
+        if (session->mSession->IsPendingEviction()) {
+            continue;
+        }
+
+        ChipLogProgress(SecureChannel, "Candidate Session[%p] - Attempting to evict...", session->mSession);
+        
+        auto prevCount = mEntries.Allocated();
+
+        //
+        // SessionHolders act like weak-refs on a session, but since they do still add to the ref-count of a SecureSession, we cannot actually
+        // tell whether there are truly any strong-refs (SessionHandles) on this session because if we did, we'd avoid evicting it since it's
+        // pointless to do so.
+        //
+        // However, we don't actually have SessionHolders implemented correctly as weak-refs, requiring us to go ahead and 'try' to evict it,
+        // and see if it still remains in the table. If it does, we have to try the next one. If it doesn't, we know we've earned a free spot.
+        //
+        // See #19495.
+        //
+        session->mSession->MarkForEviction();
+        
+        auto newCount = mEntries.Allocated();
+
+        if (newCount < prevCount) {
+            ChipLogProgress(SecureChannel, "Successfully evicted a session!");
+            mRunningEvictionLogic = false;
+            auto *retSession  = mEntries.CreateObject(*this, secureSessionType, localSessionId);
+            VerifyOrDie(session != nullptr);
+            return retSession;
+        }
+    }
+
+    mRunningEvictionLogic = false;
+    VerifyOrDieWithMsg(true, SecureChannel, "We couldn't find any session to evict at all, something's wrong!");
+    return nullptr;
+}
+
+void SecureSessionTable::DefaultEvictionPolicy(EvictionPolicyContext &evictionContext)
+{
+    evictionContext.Sort([](const auto &a, const auto &b) {
+        int aStateScore = 0, bStateScore = 0;
+
+        auto assignStateScore = [](auto &score, const auto &session) {
+            if (session.IsDefunct()) {
+                score = 2;
+            }
+            else if (session.IsActiveSession()) {
+                score = 1;
+            }
+            else {
+                score = 0;
+            }
+        };
+
+        assignStateScore(aStateScore, *a.mSession);
+        assignStateScore(bStateScore, *b.mSession);
+
+        return ((aStateScore > bStateScore) ? true : (a->GetLastActivityTime() < b->GetLastActivityTime()));
+    });
+}
+
+Optional<SessionHandle> SecureSessionTable::FindSecureSessionByLocalKey(uint16_t localSessionId)
+{
+    SecureSession * result = nullptr;
+    mEntries.ForEachActiveObject([&](auto session) {
```

Comment:
Need to pick from most recently used.

Reply:
No! The local key is guaranteed to be unique in a `SessionManager`, so there must not be conflict at all.

Reply:
Placed this comment on the wrong function (was meaning to place this on `SessionManager::FindSecureSessionForNode`).